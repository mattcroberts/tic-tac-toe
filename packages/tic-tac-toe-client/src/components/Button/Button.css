.root {
    font-size: 18px;
    font-family: inherit;
    padding: 0;
}

.small {
    height: 30px;
    padding: 0 6px;
}

.medium {
    height: 50px;
    padding: 0 12px;
}
