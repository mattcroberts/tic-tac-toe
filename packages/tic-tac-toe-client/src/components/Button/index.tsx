import * as React from "react";
import classNames from "classnames";
import style from "./Button.css";

export enum Size {
    SMALL = "small",
    MEDIUM = "medium"
}

const Button = ({
    children,
    disabled = false,
    onClick = () => undefined,
    className = "",
    size = Size.MEDIUM
}: {
    children?: any;
    disabled?: boolean;
    onClick?: () => any;
    className?: string;
    size?: Size;
}) => (
    <button
        className={classNames(style.root, style[size], className)}
        onClick={onClick}
        disabled={disabled}
    >
        {children}
    </button>
);

export default Button;
