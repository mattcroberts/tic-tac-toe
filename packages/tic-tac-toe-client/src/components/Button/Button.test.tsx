import * as React from "react";
import { shallow } from "enzyme";
import Button from ".";

describe("Button", () => {
    const defaultProps = {};

    it("should render", () => {
        const component = shallow(<Button {...defaultProps} />);
        expect(component).toMatchSnapshot();
    });

    it("should render disabled", () => {
        const component = shallow(<Button disabled={true} />);
        expect(component).toMatchSnapshot();
    });

    it("should render children", () => {
        const component = shallow(<Button>Button Text</Button>);
        expect(component).toMatchSnapshot();
    });

    it("should fire click handler", () => {
        const props = {
            ...defaultProps,
            onClick: jest.fn()
        };

        const component = shallow(<Button {...props} />);
        component.simulate("click");

        expect(props.onClick).toHaveBeenCalledTimes(1);
    });
});
