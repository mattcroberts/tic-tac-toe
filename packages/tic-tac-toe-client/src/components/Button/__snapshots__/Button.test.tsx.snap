// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<button
  className="root medium"
  disabled={false}
  onClick={[Function]}
/>
`;

exports[`Button should render children 1`] = `
<button
  className="root medium"
  disabled={false}
  onClick={[Function]}
>
  Button Text
</button>
`;

exports[`Button should render disabled 1`] = `
<button
  className="root medium"
  disabled={true}
  onClick={[Function]}
/>
`;
